### PR TITLE
[freerdp] Update to 3.0.0

### PR DESCRIPTION
**Pull request by dg0yt**
Based on #35654, CC @MehdiChinoune.

Given that this is a major update, I would prefer to switch to upstream's chosen directory for installing headers. In particular that the current patching applies it consistently (for freerdp and winpr, but not for rdtk and uwac). But this kind of normalization might be too radical for msbuild users.

### CODE REVIEW

File: ports/freerdp/vcpkg.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "license": "Apache-2.0",
   "supports": "!uwp & !(arm & windows)",
   "dependencies": [
+    "cjson",
```
Comment (MehdiChinoune) [inline ports/freerdp/vcpkg.json:9]: cjson is required for AAD (WITH_AAD=ON)

Comment (dg0yt) [reply ports/freerdp/vcpkg.json:9]: I was just looking at this. I planned to turn it on, but it needs more work on CMake config and pkgconfig. They really don't care about static linkage.

**Comment by MonicaLiu0311** [comment]
An error occurred while I testing features:
```
monica@monica003:/mnt/vcpkg$ ./vcpkg install freerdp[wayland]
Computing installation plan...
The following packages will be built and installed:
    freerdp[core,wayland]:x64-linux@2.11.2#1
Detecting compiler hash for triplet x64-linux...
Restored 0 package(s) from /home/monica/.cache/vcpkg/archives in 16.7 us. Use --debug to see more details.
Installing 1/1 freerdp[core,wayland]:x64-linux@2.11.2#1...
Building freerdp[core,wayland]:x64-linux@2.11.2#1...
-- Using cached FreeRDP-FreeRDP-2.11.2.tar.gz.
-- Cleaning sources at /mnt/vcpkg/buildtrees/freerdp/src/2.11.2-882d1e3a25.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /mnt/vcpkg/downloads/FreeRDP-FreeRDP-2.11.2.tar.gz
-- Applying patch DontInstallSystemRuntimeLibs.patch
-- Applying patch fix-include-path.patch
-- Applying patch fix-install-destination.patch
-- Applying patch keep-dup-libs.patch
-- Applying patch wfreerdp-server-cli.patch
-- Applying patch pr-7060-jni-onload.patch
-- Applying patch find-dependency.patch
-- Applying patch export-include.patch
-- Using source at /mnt/vcpkg/buildtrees/freerdp/src/2.11.2-882d1e3a25.clean
-- Configuring x64-linux
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Fixing pkgconfig file: /mnt/vcpkg/packages/freerdp_x64-linux/lib/pkgconfig/freerdp-client2.pc
-- Fixing pkgconfig file: /mnt/vcpkg/packages/freerdp_x64-linux/lib/pkgconfig/freerdp2.pc
-- Fixing pkgconfig file: /mnt/vcpkg/packages/freerdp_x64-linux/lib/pkgconfig/winpr2.pc
-- Fixing pkgconfig file: /mnt/vcpkg/packages/freerdp_x64-linux/debug/lib/pkgconfig/freerdp-client2.pc
-- Fixing pkgconfig file: /mnt/vcpkg/packages/freerdp_x64-linux/debug/lib/pkgconfig/freerdp2.pc
-- Fixing pkgconfig file: /mnt/vcpkg/packages/freerdp_x64-linux/debug/lib/pkgconfig/winpr2.pc
CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:36 (message):
  Couldn't find tool "wlfreerdp":

      "/mnt/vcpkg/packages/freerdp_x64-linux/bin/wlfreerdp" does not exist
Call Stack (most recent call first):
  ports/freerdp/portfile.cmake:105 (vcpkg_copy_tools)
  scripts/ports.cmake:170 (include)
```

**Comment by MehdiChinoune** [comment]
@MonicaLiu0311 It seems you are not using this branch. This PR is to update freerdp to 3.0.0 but your log shows 2.11.2

**Comment by MonicaLiu0311** [comment]
> @MonicaLiu0311 It seems you are not using this branch. This PR is to update freerdp to 3.0.0 but your log shows 2.11.2

Thanks for your correction, comment has been updated.

**Comment by dg0yt** [comment]
Some dependencies are not properly controlled and checked.

**Comment by dg0yt** [comment]
Updated for better control.  
Note that you will still need some additional system packages. The wayland feature needs wayland-scanner, wayland-client, wayland-cursor and xkbcommon (pc files).

**Comment by BillyONeal** [comment]
Sorry I broke the build lab Sunday night

**Comment by MonicaLiu0311** [comment]
> Updated for better control. Note that you will still need some additional system packages. The wayland feature needs wayland-scanner, wayland-client, wayland-cursor and xkbcommon (pc files).


I installed it successfully on Windows, but when I install it on two Linux machines, I still get this error: https://github.com/microsoft/vcpkg/pull/35727#issuecomment-1859630202. Some additional system packages are also installed:
```
monica@vliumonica003:/mnt/freerdp$ sudo apt-get install wayland-protocols libwayland-client0 libwayland-cursor0 libxkbcommon0
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libxkbcommon0 is already the newest version (1.4.0-1).
wayland-protocols is already the newest version (1.25-1).
libwayland-client0 is already the newest version (1.20.0-1ubuntu0.1).
libwayland-cursor0 is already the newest version (1.20.0-1ubuntu0.1).
0 upgraded, 0 newly installed, 0 to remove and 34 not upgraded.
```

Maybe I did something less? If so, please correct me.

**Comment by dg0yt** [comment]
Now also disabling kerberos.

**Comment by MonicaLiu0311** [comment]
Features `server,urbdrc,winpr-tools` are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```
Features `ffmpeg,server,urbdrc,winpr-tools,x11` are tested successfully in the following triplet:
```
x64-linux
```

After installing the package `libwayland-dev`, feature `wayland` is tested successfully in the following triplet:
```
x64-linux
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
@dg0yt Thanks for the update!
@MonicaLiu0311 Thanks for the testing!

**Pull request merged**